docs/ConcurrentStreams.hs

```diff
@@ -723,18 +723,21 @@ import Control.Monad.Trans.Class   (MonadTrans (lift))
 -- 'fromZipAsync' type combinator can be used to switch to parallel applicative
 -- zip composition:
 --
--- This takes 7 seconds to zip, which is max (1,3) + max (2,4) because 1 and 3
--- are produced concurrently, and 2 and 4 are produced concurrently:
+-- This takes 6 seconds to zip, which is max (2 + 4, 3 + 1) because both the
+-- streams are run concurrently but the elements in the stream are produced
+-- serially. There is intra-stream concurrency but no inter-stream
+-- concurrency. Intra-stream concurrency is due to 'fromZipAsync' whereas the
+-- absence of inter-stream concurrency is due to 'fromSerial'.
```

Comment:
Let's avoid using the inter/intra stream terms it may be confusing. Also, keep it terse but clear. How about: 

"The actions within each stream are executed serially, however, the two streams are run concurrently with respect to each other. Therefore, it takes 6 seconds to zip the two streams, which is the maximum of the time to evaluate stream s1 (2 + 4 = 6 seconds) and stream s2 (3 + 1 = 4 seconds)."